### PR TITLE
Self-assembling NTP Zone

### PR DESCRIPTION
**Pull request by smklein**
Fixes https://github.com/oxidecomputer/omicron/issues/2885

### CODE REVIEW

File: smf/ntp/manifest/manifest.xml [hunk 1]
```diff
@@ -67,29 +67,24 @@
...
-                     type="astring"
-                     value="" />
-            <propval name="allow"
-                     type="astring"
-                     value="" />
+        <property_group name="config" type="application">
+            <propval name='all_links' type='astring' value='unknown' />
+            <propval name='datalink' type='astring' value='unknown' />
+            <propval name='gateway' type='astring' value='unknown' />
+            <propval name='listen_addr' type='astring' value='unknown' />
```
Comment (citrus-it) [inline smf/ntp/manifest/manifest.xml:83]: These could be `type='net_address_v6'` (assuming they are always IPv6 addresses)

File: smf/ntp/manifest/manifest.xml [hunk 2]
```diff
@@ -14,7 +14,7 @@
              type="service"
              version="1">
 
-        <create_default_instance enabled="false" />
+        <create_default_instance enabled='true' />
```
Comment (citrus-it) [inline smf/ntp/manifest/manifest.xml:17]: If you're changing to single quotes here, perhaps make it consistent throughout?

**Review by citrus-it — COMMENTED** [review]
I had a few comments on the manifest and script, but mostly notes or ones you can take or leave (github doesn't seem to have a way to flag each one that way).

**Comment by karencfv** [comment]
Hey there! I'm taking over this bit of work from @smklein :) 

@citrus-it, I think I've addressed your comments, but so much has changed since your last review that it's probably a good idea to go over it again if you don't mind?

:crossed_fingers: the integration tests pass after the giant merge.

**Comment by karencfv** [comment]
I'm not certain we still need the `all_links` property anymore since `&installed_zone.links()` is not returning anything. I'm testing to see what happens if I remove it.

**Comment by davepacheco** [comment]
I'm not sure how close this is to landing, but I've rewritten quite a lot of this code in #4466.  I think it would be a lot easier to land this PR after #4466 (and I'd be happy to help with the merge) than the other way around, if that's okay.

**Comment by karencfv** [comment]
> I'm not sure how close this is to landing, but I've rewritten quite a lot of this code in https://github.com/oxidecomputer/omicron/pull/4466. I think it would be a lot easier to land this PR after https://github.com/oxidecomputer/omicron/pull/4466 (and I'd be happy to help with the merge) than the other way around, if that's okay.

@davepacheco of course! This has been open for a while anyway. I'll be on stand-by then

**Comment by karencfv** [comment]
Superseded by: https://github.com/oxidecomputer/omicron/pull/5168

This PR was really far behind, it was easier to just open up a new one

**Pull request closed without merging**